src/Pages/SignUp/SignUp.js

```diff
@@ -1,8 +1,330 @@
 import React, { Component } from "react";
+import DaumPostcode from "react-daum-postcode";
+import "./SignUp.scss";
 
 class SignUp extends Component {
+  state = {
+    id: 1,
+    emailValue: "",
+    passwordValue: "",
+    rePasswordValue: "",
+    name: "",
+    nickname: "",
+    phoneVlaue: "",
+    addressValue: "",
+    isSubmitClicked: false,
+    zoneCode: "",
+    fullAddress: "",
+    isDaumPost: false,
+    isRegister: false,
+  };
+
+  handleInputChange = e => {
+    this.setState({
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  handleOpenPost = () => {
+    this.setState({
+      isDaumPost: true,
+    });
+  };
+  handleAddress = data => {
+    let AllAddress = data.address;
+    let extraAddress = "";
+    let zoneCodes = data.zonecode;
+    if (data.addressType === "R") {
+      if (data.bname !== "") {
+        extraAddress += data.bname;
+      }
+      if (data.buildingName !== "") {
+        extraAddress +=
+          extraAddress !== "" ? `, ${data.buildingName}` : data.buildingName;
+      }
+      AllAddress += extraAddress !== "" ? ` (${extraAddress})` : "";
+    }
+    this.setState({
+      fullAddress: AllAddress,
+      zoneCode: zoneCodes,
+    });
+  };
+
+  onCheck = e => {
+    e.preventDefault();
+    const emailCheck = /^[A-Za-z0-9][A-Za-z0-9._-]+[@]{1}[a-z]+[.]{1}[a-z]{1,4}$/;
+    console.log(this.state.passwordValue.length);
+    if (this.state.emailValue === "") {
+      alert("이메일을 입력 해 주세요");
+      return;
+    }
+    if (!emailCheck.test(this.state.emailValue)) {
+      alert("이메일을 형식에 맞게 작성해 주세요");
+      return;
+    }
+    if (!this.state.passwordValue) {
+      alert("패스워드를 입력 하시오");
+      return;
+    }
+    if (!this.state.rePasswordValue) {
+      alert("비밀번호 확인을 입력 해 주세요");
+      return;
+    }
+    if (this.state.passwordValue.length < 10) {
+      alert("비밀번호 길이를 10자리 이상으로 입력해 주세요");
+      return;
+    }
+    this.signUpFinish();
+  };
+  signUpFinish = () => {
+    const {
+      emailValue,
+      passwordValue,
+      name,
+      phoneVlaue,
+      addressValue,
+    } = this.state;
+    //    console.log("회원가입 정보 전송");
+    fetch("http://10.58.2.56:8888/user/sign-up", {
+      method: "POST",
+      body: JSON.stringify({
+        email: emailValue,
+        password: passwordValue,
+        name: name,
+        phoneVlaue: phoneVlaue,
+        addressValue: addressValue,
+      }),
+    })
+      .then(res => res.json())
+      .then(res => {
+        console.log("연결");
+        if (res.message === "SUCCESS") {
+          alert("회원가입 성공");
+          this.props.history.push("/login");
+        } else {
+          alert("필수사항 입력 부탁");
+        }
+      });
+  };
   render() {
-    return <div></div>;
+    const { isDaumPost, fullAddress, zoneCode } = this.state;
+    const width = 595;
+    const height = 450;
```

Comment:
변하지 않는 데이터 값이라면 render함수 안에 넣기보다는 class밖에 빼셔서 상수로 선언하며 대문자 snake_case로 상수명을 설정합니다. 
```suggestion
    const WIDTH = 595;
    const HEIGHT = 450;
```

Reply:
넵 참고 하겠습니다 😀